circuits/src/crypto/pk_trbfv/pk_trbfv.nr

```diff
@@ -0,0 +1,102 @@
+use crate::crypto::pk_trbfv::constants::{
+    EEK_BOUND, L, N, QIS, R1_LOW_BOUNDS, R1_UP_BOUNDS, R2_BOUNDS, SIZE, SK_BOUND, TAG,
+};
+use crate::math::polynomial::Polynomial;
+use poseidon::poseidon2::Poseidon2;
+use safe::safe::SafeSponge;
+//the formulas are in the RNS mode
+pub struct BfvPublicKeyCircuit {
+    a: [Polynomial<N>; L],
+    eek: Polynomial<N>,
+    sk: Polynomial<N>,
+    r1is: [Polynomial<2 * N - 1>; L],
+    r2is: [Polynomial<2 * N - 1>; L],
+    pk0is: [Polynomial<N>; L],
+    pk1is: [Polynomial<N>; L],
+}
+
+impl BfvPublicKeyCircuit {
+    pub fn new(
+        a: [Polynomial<N>; L],
```

Comment:
I fixed these at #5 , but I guess `A` should be `Polynomial<N>` instead of `[Polynomial<N>; L]`

Reply:
Oh, maybe this is not the case. I think I did a mistake.